daisy/instance.go

```diff
@@ -215,15 +215,19 @@ func (i *Instance) setSourceMachineImage(machineImage string) {}
 func (i *Instance) register(name string, s *Step, ir *instanceRegistry, errs DError) {
 	// Register disk attachments.
 	for _, d := range i.Disks {
-		dName := d.Source
+		diskName := d.Source
 		if d.InitializeParams != nil {
 			parts := NamedSubexp(diskTypeURLRgx, d.InitializeParams.DiskType)
 			if parts["disktype"] == "local-ssd" {
 				continue
 			}
-			dName = d.InitializeParams.DiskName
+			diskName = d.InitializeParams.DiskName
 		}
-		errs = addErrs(errs, ir.w.disks.regAttach(dName, name, d.Mode, s))
+		deviceName := d.DeviceName
+		if deviceName == "" {
+			deviceName = diskName
+		}
```

Comment:
Should this logic maybe go to regAttach in case it's called from other places in the code base?

Reply:
That's a great suggestion, done